src/main/java/edu/ucar/unidata/rosetta/service/FileParserManagerImpl.java

```diff
@@ -46,27 +48,36 @@ public void setParsedFileData(List<List<String>> parsedFileData) {
 
     /**
      * A simple method that reads each line of a file, appends a new line
-     * character and appends to a StringBuffer, and returns the StringBuffer
-     * string value. This method is used to parse the file data when no header
-     * lines have been specified. TODO: refactor to return JSON
+     * character & adds to a List. The list is then turned into a JSON string.
      *
      * @param filePath The path to the file on disk.
-     * @return A String of the file data parsed by line.
+     * @return A JSON String of the file data parsed by line.
+     * @throws IOException For any file I/O or JSON conversions problems.
      */
-    public String parseByLine(String filePath) {
-        StringBuffer stringBuffer = new StringBuffer();
-        String currentLine;
-        try (BufferedReader reader = new BufferedReader(new FileReader(filePath))) {
-            while ((currentLine = reader.readLine()) != null) {
+    public String parseByLine(String filePath) throws IOException {
+        List<String> fileContents = new ArrayList<>();
+
+        FileReader fileReader = null;
+        BufferedReader bufferedReader = null;
+
+        try {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Change this "try" to a try-with-resources. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2093)